internal/core/src/index/ScalarIndexSort-inl.h

```diff
@@ -307,4 +313,20 @@ ScalarIndexSort<T>::Reverse_Lookup(size_t idx) const {
     auto offset = idx_to_offsets_[idx];
     return data_[offset].a_;
 }
+
+template <typename T>
+inline bool
+ScalarIndexSort<T>::outOfRange(const T value) {
+    auto lower = data_.begin();
+    auto upper = data_.end();
+    return value < lower->a_ || value > upper->a_;
```

Comment:
`end()` points to the next of the last, so it usually points to an invalid data. `rbegin()` is what you need

Reply:
changed